### PR TITLE
fix(docgen): Detect non-workspace package.json files as potential root

### PR DESCRIPTION
**Pull request by ciscorn**
Fix the `get_root_path_url` function to handle `package.json` files _without the `workspaces` field_ as potential root paths.

This PR fixes the current behavior to function correctly, but I think the Issue #78 should be addressed in a separate PR.

### CODE REVIEW

File: packages/svelte-docgen/src/shared.js [hunk 1]
```diff
@@ -145,38 +145,50 @@ export function get_sources(declarations, root_path_url) {
...
-	let package_json_filepath;
-	while (directory && directory !== root) {
+	let found_dir;
+	while (directory) {
 		// Case 1: pnpm workspace
-		const pnpm_workspace_filepath = path.join(directory, "pnpm-workspace.yaml");
-		if (sys.fileExists(pnpm_workspace_filepath)) {
-			return new URL(`file://${directory}`);
+		const pnpm_workspace_yaml_filepath = pathe.join(directory, "pnpm-workspace.yaml");
+		const pnpm_workspace_yml_filepath = pathe.join(directory, "pnpm-workspace.yml");
```
Comment (xeho91) [inline packages/svelte-docgen/src/shared.js:165]:
```suggestion
```

Nit: As per https://pnpm.io/10.x/workspaces

Also I haven't found in their repository codebase _(using GitHub search)_ where they handle `.yml` files.

**Review by xeho91 — APPROVED** [review]
Haha, if I may be humorous for a moment. Reading this PR reminded me of this scene: https://www.youtube.com/watch?v=5cNF8mTSRzM

File: packages/svelte-docgen/src/shared.js (hunk 1 above)
Comment (ciscorn) [reply packages/svelte-docgen/src/shared.js]: Thank you. I was overthinking it...

**Comment by ciscorn** [comment]
We might need some applications to detect this kind of bug. `apps/docs` could be one of them.

**Pull request merged**